src/Cake.Common/Tools/Chocolatey/ApiKey/ChocolateyApiKeySetter.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Globalization;
+using Cake.Core;
+using Cake.Core.IO;
+
+namespace Cake.Common.Tools.Chocolatey.ApiKey
+{
+    /// <summary>
+    /// The Chocolatey package pinner used to pin Chocolatey packages.
+    /// </summary>
+    public sealed class ChocolateyApiKeySetter : ChocolateyTool<ChocolateyApiKeySettings>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChocolateyApiKeySetter"/> class.
+        /// </summary>
+        /// <param name="fileSystem">The file system.</param>
+        /// <param name="environment">The environment.</param>
+        /// <param name="processRunner">The process runner.</param>
+        /// <param name="globber">The globber.</param>
+        /// <param name="resolver">The Chocolatey tool resolver.</param>
+        public ChocolateyApiKeySetter(IFileSystem fileSystem, ICakeEnvironment environment,
+            IProcessRunner processRunner, IGlobber globber, IChocolateyToolResolver resolver)
+            : base(fileSystem, environment, processRunner, globber, resolver)
+        {
+        }
+
+        /// <summary>
+        /// Pins Chocolatey packages using the specified package id and settings.
+        /// </summary>
+        /// <param name="apiKey">The API key.</param>
+        /// <param name="source">The Server URL where the API key is valid.</param>
+        /// <param name="settings">The settings.</param>
+        public void Set(string apiKey, string source, ChocolateyApiKeySettings settings)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException("settings");
+            }
+
+            if (string.IsNullOrWhiteSpace(apiKey))
+            {
+                throw new ArgumentNullException("apiKey");
+            }
+
+            if (string.IsNullOrWhiteSpace(source))
+            {
+                throw new ArgumentNullException("source");
+            }
+
+            Run(settings, GetArguments(apiKey, source, settings), settings.ToolPath);
+        }
+
+        private ProcessArgumentBuilder GetArguments(string apiKey, string source, ChocolateyApiKeySettings settings)
+        {
+            var builder = new ProcessArgumentBuilder();
+
+            builder.Append("apikey");
+
+            builder.Append("-s");
```

Comment:
Nothing currently, you're good.
There's just a lot of code like:
```csharp
builder.Append("--switch");
builder.Append(switchValue);
```
Soo I wanted to introduce something like
```csharp
builder.AppendSwitch("--switch", switchValue);
```